tools/src/launcher/Main.hs

```diff
@@ -322,16 +311,25 @@ writeWindowsUpdaterRunner runnerPath = do
 
 spawnNode
     :: (FilePath, [Text], Maybe FilePath)
-    -> Shell (ProcessHandle, Async ExitCode, FilePath)
+    -> IO (ProcessHandle, Async ExitCode, FilePath)
 spawnNode (path, args, mbLogPath) = do
-    echo "Starting the node"
+    putStrLn "Starting the node"
+    -- We don't explicitly close the `logHandle` here,
+    -- but this will be done when we run the `CreateProcess` built
+    -- by proc later is `system'`:
+    -- http://hackage.haskell.org/package/process-1.6.1.0/docs/System-Process.html#v:createProcess
     (logPath, logHandle) <- case mbLogPath of
         Just lp -> do
-            mktree (FP.directory lp)
-            (lp,) <$> appendonly lp
+            createDirectoryIfMissing True (directory lp)
+            (lp,) <$> openFile lp AppendMode
         Nothing -> do
             tempdir <- liftIO (fromString <$> getTemporaryDirectory)
-            mktemp tempdir "cardano-node-output.log"
+            -- FIXME (adinapoli): `Shell` from `turtle` was giving us no-resource-leak guarantees
```

Comment:
> Am I missing something? Is it what we want here?

What we want is for `reportNodeCrash` to be able to send the full output log. If I understand `Shell` correctly, the temp file will only be deleted when the whole `Shell` is finished, and so the log will be still present when we need it:

``` haskell
clientScenario logConf node wallet updater nodeTimeout report = do
    ...
    (nodeHandle, nodeAsync, nodeLog) <- spawnNode node
    ...
             whenJust report $ \repServ -> do
                 printf ("Sending logs to "%s%"\n") (toText repServ)
                 reportNodeCrash exitCode logConf repServ nodeLog
```

However, it's true that the output log might contain some information that the node log doesn't contain (e.g. if the node dies with a segfault, it won't be written to the node log). So, let's keep the output log, okay. (And later on we can just add something to the launcher to go through temp and remove old `cardano-node-output.log` files.)